precomputed chunks loading of non-default resolution broken

I tried loading some precomputed chunks volume (`precomputed://https://open-neurodata.s3.amazonaws.com/collman/collman15v2/EM25K`) and at first all works beautifully. I wanted to do full resolution of course and selected some features, annotated closed the project. Then when loading I get the following exception:

<details><summary>traceback</summary>

```pytb


ERROR 2024-02-26 13:46:57,861 log_exception 38747 7956434944 Project could not be loaded due to the exception shown above.
Aborting Project Open Action
ERROR 2024-02-26 13:46:57,868 log_exception 38747 7956434944 Traceback (most recent call last):
  File "/.../ilastik/ilastik/shell/gui/ilastikShell.py", line 1660, in _loadProject
    self.projectManager.loadProject(hdf5File, projectFilePath, readOnly)
  File "/.../ilastik/lazyflow/utility/timer.py", line 153, in wrapper
    return func(*args, **kwargs)
  File "/.../ilastik/ilastik/shell/projectManager.py", line 465, in loadProject
    serializer.deserializeFromHdf5(self.currentProjectFile, projectFilePath, self._headless)
  File "/.../ilastik/ilastik/applets/base/appletSerializer.py", line 1149, in deserializeFromHdf5
    ss.deserialize(topGroup)
  File "/.../ilastik/ilastik/applets/pixelClassification/pixelClassificationSerializer.py", line 98, in deserialize
    super().deserialize(group)
  File "/.../ilastik/ilastik/applets/base/appletSerializer.py", line 267, in deserialize
    self._deserialize(group[self.name], self.inslot)
  File "/.../ilastik/lazyflow/utility/timer.py", line 153, in wrapper
    return func(*args, **kwargs)
  File "/.../ilastik/ilastik/applets/base/appletSerializer.py", line 618, in _deserialize
    self.inslot[index][slicing] = blockArray
  File "/.../ilastik/lazyflow/slot.py", line 1017, in __setitem__
    p[key] = value
  File "/.../ilastik/lazyflow/slot.py", line 1017, in __setitem__
    p[key] = value
  File "/.../ilastik/lazyflow/slot.py", line 1017, in __setitem__
    p[key] = value
  File "/.../ilastik/lazyflow/slot.py", line 1013, in __setitem__
    self.operator.setInSlot(self.top_level_slot, self.subindex, roi, value)
  File "/.../ilastik/lazyflow/operators/opCompressedUserLabelArray.py", line 394, in setInSlot
    self._setInSlotInput(slot, subindex, roi, new_pixels)
  File "/.../ilastik/lazyflow/operators/opCompressedUserLabelArray.py", line 416, in _setInSlotInput
    block_rois = getIntersectingRois(self.Output.meta.shape, self._blockshape, (roi.start, roi.stop))
  File "/.../ilastik/lazyflow/roi.py", line 652, in getIntersectingRois
    block_rois = [getIntersection(block_roi, roi) for block_roi in block_rois]
  File "/.../ilastik/lazyflow/roi.py", line 652, in <listcomp>
    block_rois = [getIntersection(block_roi, roi) for block_roi in block_rois]
  File "/.../ilastik/lazyflow/roi.py", line 413, in getIntersection
    assert ((stop - start) > 0).all(), "Rois do not intersect: {} and {}".format(roiA, roiB)
AssertionError: Rois do not intersect: (array([  0,  98, 192,   0]), array([ 27, 142, 198,   1])) and ([13, 142, 198, 0], [14, 142, 198, 1])
```

</details>

This happens _after_ adding annotations. If I save/load before (so after selecting some features) project is loaded, but the image locks into lowest res...

--- a/ilastik/applets/dataSelection/opDataSelection.py
+++ b/ilastik/applets/dataSelection/opDataSelection.py
@@ -833,6 +833,7 @@ class OpDataSelection(Operator):
             if data_provider.meta.scales:
                 datasetInfo.laneShape = data_provider.meta.shape
                 datasetInfo.scales = data_provider.meta.scales
+                datasetInfo.working_scale = self.ActiveScale.value
                 if datasetInfo.working_scale == DEFAULT_LOWEST_SCALE_KEY:
                     # datasetInfo.working_scale may be saved to the project file, so we want a real key here
                     # if we didn't already load one from the file.

--- a/tests/test_ilastik/test_applets/dataSelection/testOpDataSelection.py
+++ b/tests/test_ilastik/test_applets/dataSelection/testOpDataSelection.py
@@ -863,12 +863,18 @@ class TestOpDataSelection_PrecomputedChunks:
         ],
     }
 
-    def test_load_precomputed_chunks_over_http(self, graph, monkeypatch):
+    @pytest.fixture
+    def op_and_dataset(self, graph, monkeypatch) -> (OpDataSelection, MultiscaleUrlDatasetInfo):
         _ = MockRemoteDataset(monkeypatch, self.MOCK_DATASET_URL, self.INFO_JSON, self.CHUNKS)
         op = OpDataSelection(graph=graph)
         op.WorkingDirectory.setValue(os.getcwd())
         op.ActiveScale.setValue("")
-        op.Dataset.setValue(MultiscaleUrlDatasetInfo(url=self.MOCK_DATASET_URL))
+        datasetInfo = MultiscaleUrlDatasetInfo(url=self.MOCK_DATASET_URL)
+        op.Dataset.setValue(datasetInfo)
+        return op, datasetInfo
+
+    def test_load_precomputed_chunks_over_http(self, op_and_dataset):
+        op, _ = op_and_dataset
         loaded_scale0 = op.Image[:].wait()
         assert numpy.allclose(loaded_scale0, self.IMAGE_SCALED)
 
@@ -876,6 +882,12 @@ class TestOpDataSelection_PrecomputedChunks:
         op.ActiveScale.setValue(scale_keys[1])
         loaded_scale1 = op.Image[:].wait()
         assert numpy.allclose(loaded_scale1, self.IMAGE_ORIGINAL)
+
+    def test_scale_updates_dataset_info(self, op_and_dataset):
+        op, datasetInfo = op_and_dataset
+        scale_keys = list(op.Image.meta.scales.keys())
+        op.ActiveScale.setValue(scale_keys[1])
+        assert datasetInfo.working_scale == scale_keys[1]
 
 
 class TestOpDataSelection_DatasetInfo:

